girder-dandi-archive/girder_dandi_archive/rest.py

```diff
@@ -102,6 +105,49 @@ def list_dandisets(self, limit, offset, sort):
         drafts = get_or_create_drafts_collection()
         return Folder().find({"parentId": drafts["_id"]}, limit=limit, offset=offset, sort=sort)
 
+    @access.public
+    @autoDescribeRoute(
+        Description("Search Dandisets")
+        .param("search", "Search Query", paramType="path")
+        .pagingParams(defaultSort="meta.dandiset.identifier")
+    )
+    def search_dandisets(self, search, limit, offset, sort):
+        # Ensure we are only looking for drafts collection child folders.
+        drafts = get_or_create_drafts_collection()
+        # TODO Currently only searching identifier, name, and description of public dandisets
+        if not search:
+            # Empty search string should return all possible results
+            return Folder().find({"parentId": drafts["_id"]}, limit=limit, offset=offset, sort=sort)
+        return Folder().find(
+            {
+                "$and": [
```

Comment:
All MongoDB query terms are implicitly `$and` together, so this shouldn't be necessary. The top level keys can be just `parentId` and `$or`.